README.md

```diff
@@ -694,7 +694,11 @@ knpu_oauth2_client:
             type: azure
             # add and set these environment variables in your .env files
             client_id: '%env(OAUTH_AZURE_CLIENT_ID)%'
+            # client_secret is optional if you use a client certificate
             client_secret: '%env(OAUTH_AZURE_CLIENT_SECRET)%'
+            # Using a client certificate requires thenetworg/oauth2-azure > 2.1.1:
+            # client_certificate_private_key: '%env(OAUTH_AZURE_CLIENT_CERTIFICATE_PRIVATE_KEY)%'
```

Comment:
for custom parameters I would suggest implement some `client_extra_params`, because every provider could have some custom implementation.

Reply:
Or maybe some option to pass a client configured!

Reply:
Why is this different from all the other custom parameters

Reply:
IMO we we are just growing and growing this config!

for example

provider A, B, C, D ...: need the config
 - X1
 - X2
 - X3
 
 provider W: need the config
 - X4
 - X5
 - X6
 
  provider Y: need the config
 - X7
 - X8
 - X9
 
 
 most of the providers needs only the configs X1,2,3 But because few providers I need to add more and more config,
 I don't see it as a good approach!